etl/command.py

```diff
@@ -301,21 +300,34 @@ def main(
     if grapher:
         sanity_check_db_settings()
 
-    dag = construct_dag(dag_path, private=private, grapher=grapher, export=export)
+    # Load DAG
+    dag = load_dag(dag_path)
+
+    # Create full DAG
+    full_dag = construct_full_dag(dag)
+
+    # Filter to the subdag based on includes & excludes
+    subdag = construct_subdag(
+        full_dag,
+        includes=includes,
+        excludes=excludes,
+        grapher=grapher,
+        export=export,
+        private=private,
+        only=only,
+        exact_match=exact_match,
+    )
 
-    excludes = exclude.split(",") if exclude else []
+    # Compile subdag into Step objects
+    steps = compile_steps(full_dag, subdag)
```

Comment:
I think the comments and docstrings should clarify what's happening here. I just saw this comment in compile_steps: `# TODO: we need full DAG here to get dependencies of each step`. Instead of a TODO, this should be part of the docstring, since it explains why we need both a full and a filtered dag.

Reply:
Good point, fixed.